common/src/Microsoft.Azure.IIoT.Abstractions/src/Serializers/Extensions/SerializerEx.cs

```diff
@@ -167,16 +169,34 @@ public static class SerializerEx {
             return serializer.Deserialize<T>(reader.ReadToEnd());
         }
 
+        /// <summary>
+        /// Deserialize from validating reader
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="serializer"></param>
+        /// <param name="reader"></param>
+        /// <param name="schemaReader"></param>
+        /// <returns></returns>
+        public static T Deserialize<T>(this ISerializer serializer,
```

Comment:
We can also add `TextReader schemaReader = null` parameter to the `Deserialize()` method above on line `167` and remove this extra method. That way all methods accepting `schemaReader` would have `null` default for it. What do you think ?

Reply:
IMHO if we were to use default schemaReader which could be overriden optionally by caller of this method then I'd use optional parameter otherwise method overload. One thing which probably makes this distinction more visible is that if we do null checks in these public methods then overload method would throw error is null is provided for schemaReader but with optional we would not know the intention of the caller. Totally your call as a maintainer of code.

Reply:
I wanted to have a consistent usage of `TextReader schemaReader` parameter. The other `Deserialize()` calls in the class that have a `schemaReader` have it as an optional parameter with `null` default value. But I am also happy with keeping this as it is as well, it's a minor comment.